tests/profiles/name_test.py

```diff
@@ -654,3 +646,25 @@ def set_name(font, nameID, string):
     ttFont = TTFont(TEST_FILE("shantell/ShantellSans-Italic[BNCE,INFM,SPAC,wght].ttf"))
     set_name(ttFont, 17, "Light")
     assert_results_contain(check(ttFont), FAIL, "bad-typographicsubfamilyname")
+
+
+def test_check_name_postscript():
+    check = CheckTester(opentype_profile, "com.adobe.fonts/check/postscript_name")
+
+    # Test a font that has OK psname. Check should PASS.
+    ttFont = TTFont(TEST_FILE("source-sans-pro/OTF/SourceSansPro-Bold.otf"))
+    assert_PASS(check(ttFont), "psname-ok")
+
+    # Change the name-table string. Check should FAIL.
+    bad_ps_name = "this-is-a (bad) name".encode("utf-16-be")
+    ttFont["name"].setName(
+        bad_ps_name,
+        NameID.POSTSCRIPT_NAME,
+        PlatformID.WINDOWS,
+        WindowsEncodingID.UNICODE_BMP,
+        WindowsLanguageID.ENGLISH_USA,
+    )
+    msg = assert_results_contain(check(ttFont), FAIL, "bad-psname-entries")
+    assert "PostScript name does not follow requirements" in msg
+    assert "May contain only a-zA-Z0-9 characters and a hyphen" in msg
+    assert "May contain not more than a single hyphen" in msg
```

Comment:
This code-test validates more than a single behavior all at once. For this reason it is less robust than the original code-tests it is replacing. An eventual bug in one of the aspects of the check implementation might remain hidden because the other aspect is enough to pass the code-test.